pkg/endpoints/routes_test.go

```diff
@@ -1,387 +1,523 @@
-		} else {
-			withID := strings.Replace(server.URL+route, "{name}", "fakeresource", -1)
-			fmt.Printf("With ID replaced {name} with fakeresource is %s \n", withID)
-			request = *dummyHTTPRequest(httpMethod, withID, requestBody)
-		}
-		response, err := http.DefaultClient.Do(&request)
+		jsonBytes, err := json.Marshal(resourceObject)
 		if err != nil {
-			t.Error("Response error from server:", err)
-			continue
+			t.Fatalf("Error marshalling %s resource\n", resource)
 		}
-		t.Log("Response from server:", response)
-		if postFunc != nil {
-			postFunc(t, &request, response)
+		httpRequestBody = bytes.NewReader(jsonBytes)
+	} else if httpMethod == "GET" {
+		// GET/{name} route
+		if strings.Contains(route, "{name}") {
+			// This should return nil if there is no POST fakeStub implementation
+			resourceObject := fakeStub(t, r, "POST", resource, namespace, resourceName)
+			if resourceObject == nil {
+				makeFake(t, r, resource, namespace, resourceName)
+			}
 		}
 	}
-}
-
-func makeRequestBody(t *testing.T, route string, httpMethod string) *bytes.Reader {
-	if httpMethod == http.MethodPost || httpMethod == http.MethodPut {
-		resourceType := getResourceType(route, httpMethod)
-		var resource interface{}
-		// Pass the identifier for what is being updated
-		if httpMethod != http.MethodPost {
-			// It's a PUT, we only support testing POSTs and PUTs currently
-			identifierIndex := strings.LastIndex(route, "/") + 1
-			routeToUse := route[identifierIndex:]
-			fmt.Printf("Route to use is %s for resource type %s \n", routeToUse, resourceType)
-			resource = fakeCRD(t, resourceType, routeToUse)
+	// Replace path params
+	serverRoute := fmt.Sprintf("%s%s", server.URL, route)
+	serverRoute = strings.Replace(serverRoute, "{namespace}", namespace, 1)
+	serverRoute = strings.Replace(serverRoute, "{name}", resourceName, 1)
 
-		} else {
-			// Get unique identifier - this is for a POST
-			resource = fakeCRD(t, resourceType, "")
+	httpReq := dummyHTTPRequest(httpMethod, serverRoute, httpRequestBody)
+	response, err := http.DefaultClient.Do(httpReq)
+	if err != nil {
+		t.Fatalf("Error getting response from %s: %v\n", serverRoute, err)
+	}
+	if response.StatusCode != expectedStatusCode {
+		t.Logf("%s route %s failure\n", httpMethod, route)
+		t.Fatalf("Response code %d did not equal expected %d\n", response.StatusCode, expectedStatusCode)
+	}
+	// Ensure "Content-Location" header routes to resource properly
+	if httpMethod == "POST" {
+		contentLocationSlice := response.Header["Content-Location"]
+		if len(contentLocationSlice) == 0 {
+			t.Fatalf("POST route %s did not set 'Content-Location' header", route)
 		}
-		if resource == nil {
-			return nil
+		// "Content-Location" header should be of the form /some/path/<resource>/name
+		postedResourceName := contentLocationSlice[0][strings.LastIndex(contentLocationSlice[0], "/")+1:]
+		t.Log("Posted resoure name:", postedResourceName)
```

Comment:
Nit, typo here (I've merged Vincent's PR into my own rebuild work for testing and noticed this)